hips/hip-0004.md

```diff
@@ -0,0 +1,114 @@
+
+* Charts:
+    * The handling of templates/ must not be changed
+    * Reserved directory names (e.g. crds) must not be removed
+    * No files or file types that are allowed may be changed to be disallowed
+    * If an object or file type is added to a chart, it must also be optional (see requirements.yaml in Helm 2)
+    * No files that were in the files object may be removed from the files object (in other words, a new file type may be added, but files cannot be removed from the files object just because they are instances of this new file type)
+
+* Templates (commands, functions, syntax, and variables):
+    * Template functions cannot be removed
+    * Return types of functions cannot change
+    * A template function signature is additive (can add more parameters) if and only if the change is backwards compatible. E.g. sum int64 int64 can be changed to foo int64 int64 [int64] (where the new third parameter is optional). Likewise, print string could be changed to print any as long as any accepts a string and still treats it as it did before. But sum int64 int64 int64 could not be changed to sum int64 int64.
+    * built-in directives, constants, and variables cannot be removed or changed in any way that would break an existing template
+    * constants and variables that contain structured information cannot change the format of that structured information. (Example: if a built-in variable called ".Version" returned "helm-v1.2.3" it could not be changed to return "v1.2.3")
+    * If a variable is changed to a function, it must not require any arguments or return a different return type than the original variable's value. (Example: If .Version was changed from a variable that returned v1.2.3, it can be changed to a function, but only if that function allows no parameters and when no parameters are specified, still returns in the format v1.2.3.
+    * the template syntax cannot change except when the change is directly from the Go programming language (who has made such changes during Helm's history -- viz. adding the - whitespace chomp character and changing variables to become mutable).
+    * occasionally, external template libraries or their dependencies have made breaking changes. We are now trying to avoid that.
+
+* Experimental Feature Flags:
+    * From time to time, at the discretion of the core maintainers, new "experimental" features may be hidden behind one or more feature flags. An example of this in Helm 3.0.0 was the inclusion of experimental repository support.
+    * Experimental features are not required to ensure backward compatibility for their feature set. (They cannot, however, break backward compatibility for other parts of Helm.) Thus, a release new release of an existing experimental feature may break APIs, change its chart representations, or modify its command-line flags as long as it does not break the compatibility of non-experimental features.
```

Comment:
typo: "a release new release" should be "a new release"

Reply:
Good eye!  Fixed.